### PR TITLE
Derive show and eq where possible

### PR DESCRIPTION
**Pull request by joseferben**

### CODE REVIEW

File: sihl/src/contract_email.ml [hunk 1]
```diff
@@ -2,11 +2,12 @@ type t =
   { sender : string
   ; recipient : string
   ; subject : string
-  ; text : string
-  ; html : string option
+  ; text : string [@opaque]
```
Comment (aronerben) [inline sihl/src/contract_email.ml:5]: I think this is too opinionated, some sihl users might want to print these

File: sihl/src/core_command.ml [hunk 2]
```diff
@@ -8,9 +8,10 @@ type t =
   { name : string
   ; usage : string option
   ; description : string
-  ; dependencies : Core_lifecycle.lifecycle list
-  ; fn : string list -> unit option Lwt.t
+  ; dependencies : Core_lifecycle.lifecycle list [@opaque]
```
Comment (aronerben) [inline sihl/src/core_command.ml:11]: Why is this opaque?

File: sihl/src/core_command.ml [hunk 3]
```diff
@@ -8,9 +8,10 @@ type t =
...
+  ; fn : string list -> unit option Lwt.t [@opaque]
   }
+[@@deriving show]
```
Comment (aronerben) [inline sihl/src/core_command.ml:14]: `eq` is missing here on purpose?

Comment (joseferben) [reply sihl/src/core_command.ml:14]: Yes, it's not clear how to define `eq` with functions

File: sihl/src/core_lifecycle.ml [hunk 4]
```diff
@@ -14,6 +14,7 @@ type lifecycle =
   ; start : unit -> unit Lwt.t
   ; stop : unit -> unit Lwt.t
   }
+[@@deriving show]
```
[the remark this answers is not in the document]
Comment (joseferben) [reply sihl/src/core_lifecycle.ml:17]: Yes, it's not clear how to define `eq` with functions

File: sihl/src/core_service.ml [hunk 5]
```diff
@@ -8,6 +8,7 @@ type t =
   ; commands : Core_command.t list
   ; server : bool
   }
+[@@deriving show]
```
[the remark this answers is not in the document]
Comment (joseferben) [reply sihl/src/core_service.ml:11]: Yes, it's not clear how to define `eq` with functions

File: sihl/src/core_command.ml (hunk 2 above)
Comment (joseferben) [reply sihl/src/core_command.ml:11]: Because lifecycles are not showable

File: sihl/src/contract_email.ml (hunk 1 above)
Comment (joseferben) [reply sihl/src/contract_email.ml:5]: My thinking was to be secure by default. The type is public so the content can be printed separately if that is clearly intended.

File: sihl/src/core_command.ml (hunk 3 above)
Comment (aronerben) [reply sihl/src/core_command.ml:14]: For the `Core_schedule.t` you did `fn : unit -> unit Lwt.t [@equal fun _ _ -> true]`, is this not applicable here too?

File: sihl/src/contract_email.ml (hunk 1 above)
Comment (aronerben) [reply sihl/src/contract_email.ml:5]: Good idea, good concept

File: sihl/src/core_command.ml (hunk 3 above)
Comment (joseferben) [reply sihl/src/core_command.ml:14]: That's actually a fair point!

**Pull request merged**